runhouse/resources/secrets/provider_secrets/kubeconfig_secret.py

```diff
@@ -0,0 +1,62 @@
+import copy
+import os
+from typing import Dict
+
+import yaml
+
+from runhouse.resources.secrets.provider_secrets.provider_secret import ProviderSecret
+from runhouse.resources.secrets.utils import _check_file_for_mismatches
+from runhouse.utils import create_local_dir
+
+
+class KubeConfigSecret(ProviderSecret):
+    """
+    .. note::
+        To create a KubeConfigSecret, please use the factory method :func:`provider_secret` with ``provider=="kubernetes"``.
+    """
+
+    _PROVIDER = "kubernetes"
```

Comment:
I named the class kube config to be more explicit, but should we also have the provider here be `"kubeconfig"` or leave as `"kubernetes"`? the flow to construct the secret from factory methods would be `rh.provider_secret("kubernetes/kubeconfig")`

Reply:
@dongreenberg you have a strong opinion about this? thinking ahead would there be other k8s data we might want to store?